From 8dd77b93500377089725b65beacb744b1b0c226e Mon Sep 17 00:00:00 2001
From: Dhruva Gole <d-gole@ti.com>
Date: Wed, 5 Feb 2025 14:27:56 -0600
Subject: [PATCH 05/68] TI: arm64: dts: ti: k3-am62x-sk-common: enable uart as
 wakeup source

The main_uart0 ie. the linux serial console can be used as a wakeup
source using IO Daisychain to trigger system wakeup from deep sleep
states.

Co-developed-by: Tony Lindgren <tony@atomide.com>
Signed-off-by: Tony Lindgren <tony@atomide.com>
Signed-off-by: Dhruva Gole <d-gole@ti.com>
Signed-off-by: Kendall Willis <k-willis@ti.com>
---
 arch/arm64/boot/dts/ti/k3-am62x-sk-common.dtsi | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm64/boot/dts/ti/k3-am62x-sk-common.dtsi b/arch/arm64/boot/dts/ti/k3-am62x-sk-common.dtsi
index 2f129e8cd5b9..5c5cde664c6e 100644
--- a/arch/arm64/boot/dts/ti/k3-am62x-sk-common.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-am62x-sk-common.dtsi
@@ -332,6 +332,9 @@ &main_uart0 {
 	status = "okay";
 	pinctrl-names = "default";
 	pinctrl-0 = <&main_uart0_pins_default>;
+	interrupts-extended = <&gic500 GIC_SPI 178 IRQ_TYPE_LEVEL_HIGH>,
+			       <&main_pmx0 0x1c8>; /* (D14) UART0_RXD PADCONFIG114 */
+	interrupt-names = "irq", "wakeup";
 };
 
 &main_uart1 {
-- 
2.47.2

